src/hint_processor/builtin_hint_processor/vrf/inv_mod_p_uint512.rs

```diff
@@ -0,0 +1,213 @@
+use core::ops::Shl;
+
+use crate::stdlib::prelude::String;
+use crate::stdlib::vec::Vec;
+use crate::{
+    hint_processor::{
+        builtin_hint_processor::hint_utils::get_relocatable_from_var_name,
+        hint_processor_definition::HintReference,
+    },
+    math_utils::div_mod,
+    serde::deserialize_program::ApTracking,
+    stdlib::collections::HashMap,
+    vm::errors::hint_errors::HintError,
+};
+use felt::Felt252;
+use num_bigint::{BigInt, BigUint};
+use num_traits::{Num, One};
+
+use crate::vm::vm_core::VirtualMachine;
+
+/*
+def pack_512(d0, d1,d2,d3, num_bits_shift: int) -> int:
+    limbs = (d0, d1, d2, d3)
+    return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))
+
+*/
+fn pack_512(limbs: &[Felt252; 4], num_bits_shift: usize) -> BigUint {
+    limbs
+        .iter()
+        .enumerate()
+        .map(|(idx, value)| value.to_biguint().shl(idx * num_bits_shift))
+        .sum()
+}
+
+/*
+Implements hint:
+%{
+    def pack_512(u, num_bits_shift: int) -> int:
+        limbs = (u.d0, u.d1, u.d2, u.d3)
+        return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))
+
+    x = pack_512(ids.x, num_bits_shift = 128)
+    p = ids.p.low + (ids.p.high << 128)
+    x_inverse_mod_p = pow(x,-1, p)
+
+    x_inverse_mod_p_split = (x_inverse_mod_p & ((1 << 128) - 1), x_inverse_mod_p >> 128)
+
+    ids.x_inverse_mod_p.low = x_inverse_mod_p_split[0]
+    ids.x_inverse_mod_p.high = x_inverse_mod_p_split[1]
+%}
+*/
+pub fn inv_mod_p_uint512(
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let limbs_ptr = get_relocatable_from_var_name("x", vm, ids_data, ap_tracking)?;
+    let limbs: Vec<Felt252> = vm
+        .get_integer_range(limbs_ptr, 4)?
+        .iter()
+        .map(|f| f.clone().into_owned())
+        .collect();
+
+    let x = pack_512(
+        &limbs
+            .try_into()
+            .map_err(|_| HintError::FixedSizeArrayFail(4))?,
+        128,
+    );
+
+    let p_ptr = get_relocatable_from_var_name("p", vm, ids_data, ap_tracking)?;
+    let p_low = vm.get_integer(p_ptr)?;
+    let p_high = vm.get_integer((p_ptr + 1_i32)?)?;
+
+    let p = p_low.into_owned().to_biguint() + (p_high.into_owned().to_biguint() << 128_usize);
+    let x_inverse_mod_p =
+        Felt252::from(div_mod(&BigInt::one(), &BigInt::from(x), &BigInt::from(p)));
+
+    let x_inverse_mod_p_ptr =
+        get_relocatable_from_var_name("x_inverse_mod_p", vm, ids_data, ap_tracking)?;
+
+    vm.insert_value(
+        x_inverse_mod_p_ptr,
+        &x_inverse_mod_p
+            & &Felt252::from_str_radix("340282366920938463463374607431768211455", 10).unwrap(),
```

Comment:
To make it easier to understand where the number comes from, we could change it to:
```suggestion
            & &Felt252::from(u128::MAX),
```

Reply:
Thanks tommy!